### PR TITLE
MOBILE-85 Wallet Connect Android Integration Part 1

### PR DESCRIPTION
**Pull request by cristhianescobar**
Adding initial integration with Wallet Connect, we still need to investigate further the values we need from each blockchain in order to query their music. Also added a refactor I had going on with Gradle files. The option to for wallet connect is only available in debug since it's not fully functional yet.

### CODE REVIEW

File: android/app-newm/build.gradle.kts [hunk 1]
```diff
@@ -1,3 +1,11 @@
+import com.android.build.gradle.internal.cxx.configure.gradleLocalProperties
+import org.jetbrains.kotlin.konan.properties.Properties
+
+apply(from = "../../gradle_include/compose.gradle")
+apply(from = "../../gradle_include/circuit.gradle")
```
Comment (cristhianescobar) [inline android/app-newm/build.gradle.kts:5]: @newmskywalker going forward, you will only need to add this line to your Gradle file to bring in circuit dependencies

Comment (newmskywalker) [reply android/app-newm/build.gradle.kts:5]: why did we have to add these?

**Review by newmskywalker — APPROVED** [review]

File: android/app-newm/src/main/java/io/newm/wallet/connect/WalletConnectInitializer.kt [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+    private fun initializeCoreClient() {
+        val connectionType = ConnectionType.AUTOMATIC
+        val relayUrl = "relay.walletconnect.com"
+        val serverUrl = "wss://$relayUrl?projectId=${BuildConfig.WALLET_CONNECT_PROJECT_ID}"
+        val appMetaData = Core.Model.AppMetaData(
+            name = "NEWM",
+            description = "Link your wallet to NEWM",
+            url = "https://newm.io/",
+            icons = listOf("https://developers.cardano.org/img/devblog/newm.png"),
```
Comment (martyu) [inline android/app-newm/src/main/java/io/newm/wallet/connect/WalletConnectInitializer.kt]: This file could get deleted right? Maybe use the one on ours. 
```suggestion
            icons = listOf("https://newm.io/wp-content/uploads/2022/06/NEWM_Logo-1536x1536.png"),
```

File: android/app-newm/src/main/java/io/newm/wallet/connect/WalletConnectInitializer.kt [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            ),
+            onSuccess = {
+                Timber.tag(TAG).d("Web3Modal initialized successfully")
+            },
+            onError = { error ->
+                Timber.tag(TAG).d("Web3Modal initialization error: $error")
+            }
+        )
+        Web3Modal.setDelegate(WalletConnectModalDelegate)
+        Web3Modal.setChains(Web3ModalChainsPresets.ethChains.values.toList())
```
Comment (martyu) [inline android/app-newm/src/main/java/io/newm/wallet/connect/WalletConnectInitializer.kt:58]: Eth chains?

File: shared/src/commonMain/kotlin/io.newm.shared/public/constants/WalletConnectConstants.kt [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package io.newm.shared.public.constants
+
+object WalletConnectConstants {
+    private const val COINBASE: String = "fd20dc426fb37566d803205b19bbc1d4096b248ac04548e3cfb6b3a38bd033aa"
+    private const val TRUST_WALLET = "4622a2b2d6af1c9844944291e5e7351a6aa24cd7b23099efac1b2fd875da31a0"
+    private const val META_MASK = "c57ca95b47569778a828d19178114f4db188b89b763c899ba0be274e97267d96"
+    private const val BLOCKCHAIN_COM = "84b43e8ddfcd18e5fcb5d21e7277733f9cccef76f7d92c836d0e481db0c70c04"
+    private const val CRYPTO_COM = "f2436c67184f158d1beda5df53298ee84abfc367581e4505134b5bcf5f46697d"
+    private const val RAINBOW = "1ae92b26df02f0abca6304df07debccd18262fdf5fe82daa81593582dac9a369"
```
Comment (martyu) [inline shared/src/commonMain/kotlin/io.newm.shared/public/constants/WalletConnectConstants.kt:9]: Is this all we need to do to add more wallets? Just add to this list?

**Review by martyu — APPROVED** [review]

File: android/app-newm/build.gradle.kts (hunk 1 above)
Comment (cristhianescobar) [reply android/app-newm/build.gradle.kts:5]: So we don't have to specify same dependencies over and over thought the modules

**Pull request merged**